pythran/pythonic/include/types/numpy_expr.hpp

```diff
@@ -58,20 +212,47 @@ namespace pythonic
       auto fast(long i) const -> decltype(
           this->_fast(i, typename utils::gens<sizeof...(Args)>::type{}));
 
+      template <int... I>
+      auto _ufast(std::array<long, sizeof...(I)> const &indices,
+                  utils::seq<I...>) const
+          -> decltype(Op()(std::get<I>(args).fast(std::get<I>(indices))...));
+
+      template <class... Indices>
+      auto ufast(Indices... indices) const -> decltype(
+          this->_ufast(std::array<long, sizeof...(Indices)>{{indices...}},
+                       typename utils::gens<sizeof...(Args)>::type{}));
+
+    public:
+      // FIXME: this does not take into account bounds and broadcasting
       auto operator[](long i) const -> decltype(this->fast(i));
       auto operator()(long i) const -> decltype(this->fast(i));
 
       array<long, value> const &shape() const;
 
 #ifdef USE_BOOST_SIMD
+      using simd_iterator = const_simd_nditerator<numpy_expr>;
+      simd_iterator vbegin() const;
+      simd_iterator vend() const;
+
       template <int... I>
       auto _load(long i, utils::seq<I...>) const
           -> decltype(Op()(std::get<I>(args).load(i)...));
 
-      template <class I> // template to prevent automatic instantiation when the
-      // type is not vectorizable
+      // template to prevent automatic instantiation when the type is not
+      // vectorizable
+      template <class I>
       auto load(I i) const -> decltype(
           this->_load(i, typename utils::gens<sizeof...(Args)>::type{}));
+
+      template <int... I>
+      auto _uload(std::array<long, sizeof...(I)> const &indices,
```

Comment:
On Sat, May 21, 2016 at 12:57:26PM -0700, pbrunet wrote:
> >        auto load(I i) const -> decltype(
> >            this->_load(i, typename utils::gens<sizeof...(Args)>::type{}));
> > +
> > +      template <int... I>
> > +      auto _uload(std::array<long, sizeof...(I)> const &indices,
> 
> uload? Do you mean upload? It doesn't look to be use if test passes.

uload is not a very goodname, it should be map_load, I'll change this.
it's used by the numexpr iterator.